frame/nfts/src/lib.rs

```diff
@@ -224,6 +224,18 @@ pub mod pallet {
 		OptionQuery,
 	>;
 
+	#[pallet::storage]
+	/// Stores collection roles as per account.
+	pub(super) type CollectionRoleOf<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
```

Comment:
From the code below, I see that actually only 1 account per role is possible. But structure allows more.
Why not, key = Collection + Role, Value = AccountId
In this case you don't even need a bit mask

This might require more reads, if within one call we need to check different roles for the same account.